src/Reports.ReportPluginBase/ReportPluginBase.cs

```diff
@@ -65,19 +65,10 @@ public string GenerateReportIntoFile(RunFileReportRequest request)
             string outputFormat = request.OutputFormat;
             Log.DebugFormat("GenerateReport after RenderDocument, document name is {0}, page count is {1}, outputFormat is {2}",
                 document.Name, document.Pages.Count, outputFormat);
-
-            string tempFileName = Path.GetTempFileName();
+            
+            string tempFolderPath = Path.GetTempPath();
```

Comment:
Ah, I see what you mean with the TempFile issue. I guess my understanding of it was more towards the naming part but it has to do with the actual access of the file. However, because of the race condition you describe where `GetTempFileName` could return the same file to a different process, the `exportFilter.Export()` function runs into the file access error. I did some more digging with this error and found that the external library used by the reports repo (`PerpetuumSoft`) uses a `FileStream` to create and write to the file. I believe this is where the issue happens.

![image](https://github.com/AquaticInformatics/time-series-custom-reports/assets/83607090/f6698e22-c885-430e-9bb1-2299ee40ad8e)
![image](https://github.com/AquaticInformatics/time-series-custom-reports/assets/83607090/ac39083c-282c-4901-9fa3-26adc9986397)

Now that we switched to `GetRandomFileName`, we won't run into this race condition. As you mentioned and I should have just searched up (https://docs.fluidattacks.com/criteria/fixes/csharp/028/#:~:text=Additionally%2C%20the%20Path.GetTempFileName()%20function%20could%20potentially%20return%20the%20same%20filename%20to%20different%20processes%20if%20called%20simultaneously%2C%20leading%20to%20a%20race%20condition%20where%20one%20process%20could%20overwrite%20the%20sensitive%20data%20written%20by%20another.%20This%20could%20result%20in%20data%20corruption%20or%20exposure.). Using `GetRandomFileName` also removes the need to delete the file as the function doesn't create the file.